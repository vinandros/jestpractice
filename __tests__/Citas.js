import Citas from "../js/classes/Citas";

describe("Test Citas class", () => {
  const citas = new Citas();
  const id = "2";
  test("should add new Cita", () => {
    const citaObj = {
      mascota: "Hok",
      propietario: "Kevin",
      telefono: "87892400",
      fecha: "10-2-2020",
      hora: "10:12",
      sintomas: "Saludable",
    };

    citaObj.id = id;

    citas.agregarCita(citaObj);

    expect(citas).toMatchSnapshot();
  });

  test("should update cita", () => {
    const citaUpdated = {
      id,
      mascota: "Hook",
      propietario: "Kevin",
      telefono: "87892400",
      fecha: "10-2-2020",
      hora: "10:12",
      sintomas: "Saludable",
    };

    citas.editarCita(citaUpdated);

    expect(citas).toMatchSnapshot();
  });

  test("should delete Cita", () => {
    citas.eliminarCita(id);
    expect(citas).toMatchSnapshot();
  });
});
