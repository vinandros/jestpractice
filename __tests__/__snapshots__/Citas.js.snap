// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Citas class should add new Cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-2-2020",
      "hora": "10:12",
      "id": "2",
      "mascota": "Hok",
      "propietario": "Kevin",
      "sintomas": "Saludable",
      "telefono": "87892400",
    },
  ],
}
`;

exports[`Test Citas class should delete Cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Test Citas class should update cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-2-2020",
      "hora": "10:12",
      "id": "2",
      "mascota": "Hook",
      "propietario": "Kevin",
      "sintomas": "Saludable",
      "telefono": "87892400",
    },
  ],
}
`;
